apps/gallery/js/ImageEditor.js

```diff
@@ -915,6 +908,22 @@ ImageEditor.prototype.resize = function() {
   }
 };
 
+// Reset and generate new preview using the updated dimensions
+// in this.source object
+ImageEditor.prototype.requestNewPreview = function(callback) {
+  if (this.preview.src) {
+    URL.revokeObjectURL(this.preview.src);
+    this.preview.removeAttribute('src');
+    $('edit-enhance-button').classList.add('disabled');
+  }
+
+  this.edit(function() {
+    if (callback) {
+      callback();
+    }
+  });
+};
```

Comment:
I agree, will update to keep these as two separate functions resetPreview and edit, so that resetPreview can be used without calling edit every time.